packages/ra-core/src/controller/list/useSelectAll.spec.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+
+import {
+    Basic,
+    defaultDataProvider,
+    Limit,
+    QueryOptions,
+} from './useSelectAll.stories';
+
+describe('useSelectAll', () => {
+    it('should select all', async () => {
+        render(<Basic options={{}} />);
+        await screen.findByText('Selected ids: []');
+        fireEvent.click(screen.getByText('Select All'));
+        await waitFor(() => {
+            expect(screen.queryByText('Selected ids: []')).not.toBeNull();
```

Comment:
I don't understand this test.  You should check the string `Selected ids: [1,2,3,4,5,6,7]`